dbus-tokio/src/connection.rs

```diff
@@ -69,55 +72,87 @@ enum IOResourceRegistration {
 pub struct IOResource<C> {
     connection: Arc<C>,
     registration: IOResourceRegistration,
+    waker: Waker,
     write_pending: bool,
 }
 
-fn is_poll_ready(i: task::Poll<mio::Ready>) -> bool {
-    match i {
-        task::Poll::Ready(r) => !r.is_empty(),
-        task::Poll::Pending => false,
+/// An error that can occur in the dbus-tokio reactor
+#[derive(Debug)]
+pub struct IOResourceError {
+    inner: ResourceErrorKind
+}
+#[derive(Debug)]
+enum ResourceErrorKind {
```

Comment:
How about we make the enum public and skip "IOResourceError"? Potentially with an added non_exhaustive attribute.

Reply:
Also add `derive(Clone)` btw

Reply:
Alright, wasn't sure whether I should add more public API surface area, since `Box<dyn Error>` is pretty anonymous, but I guess it makes sense that a dbus IO reactor would fail with either an IO or a dbus error :)

Reply:
Oh, neither of the inner error types implement Clone.